llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -9584,7 +9743,7 @@ static bool areRuntimeChecksProfitable(GeneratedRTChecks &Checks,
   }
 
   // The scalar cost should only be 0 when vectorizing with a user specified VF/IC. In those cases, runtime checks should always be generated.
-  double ScalarC = *VF.ScalarCost.getValue();
+  uint64_t ScalarC = *VF.ScalarCost.getValue();
```

Comment:
I assume it should work even with vscale x 1. Plus, I think we'd better to use ints instead of double in the compiler. I prepare a separate patch with this fix.